src/colmap/sensor/models.h

```diff
@@ -1592,4 +1665,46 @@ double CameraModelCamFromImgThreshold(const CameraModelId model_id,
   return -1;
 }
 
+bool CameraModelIsFisheye(const CameraModelId model_id) {
+  switch (model_id) {
+#define CAMERA_MODEL_CASE(CameraModel) case CameraModel::model_id:
+
+    FISHEYE_CAMERA_MODEL_CASES
+    return true;
+    default:
+      return false;
+
+#undef CAMERA_MODEL_CASE
+  }
+
+  return false;
+}
+
+bool FisheyeCameraModelIsValidPixel(const CameraModelId model_id,
```

Comment:
I know that there have not been any cases that hit this, but I could see the potential use case of this feature outside the tests. Should we just keep it as exposed since this is a natural feature for Fisheye cameras?

Reply:
Sure sounds good. This is done now.